modules/common/condition/funcs.go

```diff
@@ -86,6 +86,21 @@ func (conditions *Conditions) Set(c *Condition) {
 	conditions.Sort()
 }
 
+// Remove a condition from the slice of conditions
+func (conditions *Conditions) Remove(conditionType Type) {
+	if conditions == nil || len(*conditions) == 0 {
+		return
+	}
+	newConditions := make([]Condition, 0, len(*conditions)-1)
```

Comment:
if the condition type is not part of the conditions list, the `newConditions` will not have the correct size, its not really an issue as `append()` will handle that.

Reply:
Yeah, it's just initial size for the slice and in normal scenario would be fine.